### DISCUSSION TITLE
Visual Studio Code IDE and micropython on STM32

### DISCUSSION THREAD

**Discussion by Bortania** [micropython/micropython#16204] [General]
Hello! 
Is it possibile to work with micropython using Visual Studio Code? I want to write script in Visual Studio Code and execute it on the STM32 board directly from Visual Studio IDE.
Thanks.

**Comment by mattytrentini** [comment]
Yes, I use VS Code daily for MicroPython work. I edit files in VS Code and then use [mpremote](https://docs.micropython.org/en/latest/reference/mpremote.html) to communicate to the device, often using the mount option. 

To execute a specific file on your device, install mpremote, use the terminal built-in to VS Code and execute something like: `mpremote run yourfile.py`.

**Reply by daleka** [reply]
> Yes, I use VS Code daily for MicroPython work. I edit files in VS Code and then use [mpremote](https://docs.micropython.org/en/latest/reference/mpremote.html) to communicate to the device, often using the mount option.
> 
> To execute a specific file on your device, install mpremote, use the terminal built-in to VS Code and execute something like: `mpremote run yourfile.py`.

Hi! I'm having trouble using mpremote with my Pyboard on COM10.

After connecting, I see this:
PS E:\Thony\Executive> mpremote connect COM10
Connected to MicroPython at COM10
Use Ctrl-] or Ctrl-x to exit this shell
or
PS E:\Thony\Executive> mpremote connect auto
Connected to MicroPython at COM10
Use Ctrl-] or Ctrl-x to exit this shell

But after this, nothing happens. The terminal does not accept any input, and it seems frozen. However, it does react to Ctrl-x or Ctrl-], which just exits the session.

I tried specifying COM10 manually and using auto, but the result is the same. I can't enter any commands in REPL, and the device does not respond.

Could mpremote require any additional dependencies? Are there any settings I should check? I would really appreciate any advice!

**Comment by bidrohini** [comment]
Yes, it is possible to work with MicroPython on Visual Studio Code (VS Code) and execute scripts directly on an STM32 board. https://lemariva.com/blog/2018/12/micropython-visual-studio-code-as-ide                                                                                                                      https://community.st.com/t5/stm32-mcus-products/can-i-use-python-to-program-stm32/td-p/177775

**Comment by Josverl** [comment]
I think that VSCode + MicroPico can support stm32 as well , ( at least its predecessor VSCode + Pymakr could)

https://github.com/paulober/MicroPico?tab=readme-ov-file#features
I know Paul has added a lot of features for the rp2 boards, not sure if that interferes with stm32 
 
and it comes with some nice type-stubs as well 😎
